core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2767,3 +2767,24 @@ private[spark] class CircularBuffer(sizeInBytes: Int = 10240) extends java.io.Ou
     new String(nonCircularBuffer, StandardCharsets.UTF_8)
   }
 }
+
+
+/**
+ * Factory for URL stream handlers. It relies on 'protocol' to choose the appropriate
+ * UrlStreamHandlerFactory to create URLStreamHandler. Adding new 'if' branches in
+ * 'createURLStreamHandler' like 'hdfsHandler' to support more protocols.
+ */
+private[spark] class SparkUrlStreamHandlerFactory extends URLStreamHandlerFactory {
+  private var hdfsHandler : URLStreamHandler = _
+
+  def createURLStreamHandler(protocol: String): URLStreamHandler = {
+    if (protocol.compareToIgnoreCase("hdfs") == 0) {
```

Comment:
I don't think you have to do compare here, Hadoop itself will find out supported FS through `fs.*.impl` and service loader.

Be noted HDI will use wasb by default, so your assumption here ("hdfs") may potentially break their codes.

This comes to the question below, why we need to wrap the `FsUrlStreamHandlerFactory` here? The only difference is that you add one more check to see if it is hdfs or not. I think it is not necessary and is handled by hadoop already.